### DISCUSSION TITLE
.net 6 and Orchard 1.5 migration

### DISCUSSION THREAD

**Discussion by RobinC4** [OrchardCMS/OrchardCore#13150] [Q&A] [answered]
Hello,

We've been using orchard core since 1.0 already and have been building out our own platform over the passed time.
Now it was decided to finally move to .net 6 (coming from .net 5) and updating the orchard nuget packages also to the latest 1.5 version to match the .net version.

Had some issues with references to some other packages but everything has been updated and upgraded to .net 6.
The solution itself builds, I can fire up the website but instead of the login screen i'm met with a bunch of errors which seem to come from the razor page (I see no migration errors in the logs).

I've looked through the release notes, and other .net 6 breaking changes and I've been trying to add and remove references if something would be missing but to no avail so far. I can run a clean .net 6 project referencing the Orchard.Application.Cms.Core.Targets no problem.

From other searches updating from 1.0 to 1.5 is some undertaking and noted to be done in steps. However, I could not find the steps unless meant to first go to 1.1, 1.2, .... So we straight went for 1.5.

Not sure if anyone has seen this before and can help me out or nudge me in the right direction because I just don't see it what is missing. For reference I've added some screenshots of the error messages I get on the page, and also added the log that's created when the application is started.

Thanks for any help.

![OC_2](https://user-images.githubusercontent.com/67279352/214832895-7cb3cfd2-f37e-49b4-a33b-583c49c93371.JPG)
![OC_3](https://user-images.githubusercontent.com/67279352/214832900-7eee0895-468f-4e28-a84d-8768ce0b8459.JPG)
![OC_1](https://user-images.githubusercontent.com/67279352/214832902-dfe10b80-d6bb-47bb-8ba4-6fa82e754075.JPG)
[orchard-log20230126.txt](https://github.com/OrchardCMS/OrchardCore/files/10509101/orchard-log20230126.txt)

**Comment by Skrypt** [comment]
This looks like a compilation issue related with the new Global Using Statements In C# 10.

**Reply by RobinC4** [reply]
Thanks for the quick reply.
C# 10 is still new for me.
But as I can understand it's a new feature and not breaking as we still have all the local usings in place as before.
In Orchard they might be used but that means they are provided within Orchard itself? Or do I have to create a using list somewhere?

I tried adding a global using list for things like system and microsoft, but the errors are still there.

**Reply by ns8482e** [reply]
.net 6 and global using are not compatible with orchard core 1.5 while razor compilation enabled

**Reply by ns8482e** [reply]
Remove razor compilation reference from project

**Comment by RobinC4** [comment, marked as answer]
Hello, thanks for the suggestion. I had already added RazorRuntimeCompilation as false and PackageReference Include="Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation" Condition="'$(RazorRuntimeCompilation)' == 'true'".

But when looking at those settings I noticed the following setting.
PreserveCompilationReferences as true

I removed this and then no more errors...
